### PR TITLE
FIX: margin calculation

### PR DESCRIPTION
**Pull request by altairis-tof**
also add total line for agent margins and remove old comments

### CODE REVIEW

**Comment by eldy** [comment]
I can't reproduce bug.
Can you describe an simple use case to show error before fix ?

**Comment by altairis-tof** [comment]
the better way to reproduce the bug is to check totals in margin reports which are wrong without this fix, especially when there are negative lines

**Comment by fastm3** [comment]
A test case which shows the error is described in #5965  . I guess it will be fixed as well.

**Comment by altairis-tof** [comment]
thanks @fastm3 
ping @eldy

File: htdocs/margin/agentMargins.php [hunk 1]
```diff
@@ -115,8 +115,8 @@
 if ($agentid > 0) $sql.= " s.rowid as socid, s.nom as name, s.code_client, s.client,";
 $sql.= " u.rowid as agent, u.login, u.lastname, u.firstname,";
 $sql.= " sum(d.total_ht) as selling_price,";
-$sql.= " sum(".$db->ifsql('d.total_ht < 0','d.qty * d.buy_price_ht * -1','d.qty * d.buy_price_ht').") as buying_price,";
-$sql.= " sum(".$db->ifsql('d.total_ht < 0','-1 * (abs(d.total_ht) - (d.buy_price_ht * d.qty))','d.total_ht - (d.buy_price_ht * d.qty)').") as marge" ;
+$sql.= " sum(".$db->ifsql('d.total_ht <0','d.qty * abs(d.buy_price_ht) * -1','d.qty * d.buy_price_ht').") as buying_price,";
```
Comment (eldy) [inline htdocs/margin/agentMargins.php:118]: Do you have value into database buy_price_ht negative into your database ? (may be on old record ?) 
All rest of code works with the following rule in mind: the field to store buy/cost price is always positive into database.

**Pull request closed without merging**